### PR TITLE
Added CONTAINER_TOOL to set container tool for operator build (#6089)

### PR DESCRIPTION
**Pull request by stanislavulrych**
Implementation of https://github.com/operator-framework/operator-sdk/issues/6089

### CODE REVIEW

**Comment by jmrodri** [comment]
@stanislavulrych if you don't mind adding a changelog for this PR. The contents of the PR looks great.

**Comment by stanislavulrych** [comment]
@jmrodri Added. Please see the last commit.

File: changelog/fragments/enable-setting-of-container-tool.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  - description: >
+      For Go-based operators: Added `CONTAINER_TOOL` variable for setting of container tool for 
+      building the operator, bundle and catalog images.
+
+      By default, docker is used. 
+      
+      Use `CONTAINER_TOOL=podman` to build with podman. The settings can be passed to the make 
+      commands by setting 'make <target> CONTAINER_TOOL=podman' or set in the `Makefile`. 
+
+      The taget names `docker-build`, `docker-push` in the `Makefile` are kept to keep backward compability.
```
Comment (everettraven) [inline changelog/fragments/enable-setting-of-container-tool.yaml:13]:
```suggestion
      The target names `docker-build`, `docker-push` in the `Makefile` are kept to keep backward compability.
```

**Comment by stanislavulrych** [comment]
@everettraven originally, I intended to only implement this change for go based operators. But from the running of `make generate`, the changes is also propagated to ansible based operator, hence further changes. I also commit the generated tests. 

However, I can still see there are docker targets hardcoded and the resulting test code is not as I expect it.

Specifically, there are targets using `docker buildx`. Unlike other docker commands, `buildx` does not have direct equivalent in podman, so this would require podman/buildah specific commands.

**Comment by everettraven** [comment]
Hi @stanislavulrych I think it is fine for the changes to go into the ansible and helm operators as well, but now that I think about it the Go operator scaffolding is done via Kubebuilder. Operator SDK pulls in Kubebuilder's CLI in and extends it to add the ansible and helm plugins but the go/v3 plugin is maintained as part of the Kubebuilder source code. 

Any changes to the scaffolds for a Go based operator would have to be done in https://github.com/kubernetes-sigs/kubebuilder (aside from the `catalog-build` and `catalog-push` targets you successfully modified here - those are maintained by us with the manifests plugin). 

If you are still interested in continuing to contribute this functionality, we can hold this PR on the necessary changes being made in Kubebuilder and then updating this PR to contain the necessary changes for SDK (I presume that most of these changes will stay the same, but I would personally rather wait until the changes are merged into Kubebuilder and pulled into SDK before we commit these changes just in case something needs to be changed slightly).

> Specifically, there are targets using docker buildx. Unlike other docker commands, buildx does not have direct equivalent in podman, so this would require podman/buildah specific commands.

This would also have to be a change made in Kubebuilder. There would likely need to be a bit more discussion as to how to handle this properly over there, but as a reviewer/maintainer in Kubebuilder I would be happy to help discuss any changes that need to be made.

**Comment by stanislavulrych** [comment]
Hi @everettraven I agree and I will try to open a respective pull request in kubebuilder then.

**Comment by everettraven** [comment]
@stanislavulrych Sounds good. I will go ahead and add a hold to this PR for the time being. Thanks  for the contributions!

/hold

**Pull request closed without merging**